### ISSUE TITLE
panic in fallback.rs:817

### ISSUE THREAD

**Issue by stef** [dtolnay/proc-macro2#419]
I'm trying to build rust-fido2 on alpine linux, when doing cargo build i get this error:

```
  thread 'main' panicked at '"pthread_attr_t_union_(unnamed_at_/usr/include/bits/alltypes_h_378_18)" is not a valid Ident', /home/s/.cargo/registry/src/index.crates.io-6f17d22bba15001f/proc-macro2-1.0.69/src/fallback.rs:817:9
```

while `alltypes.h` contains this at line 378:
```c
typedef struct { union { int __i[sizeof(long)==8?14:9]; volatile int __vi[sizeof(long)==8?14:9]; unsigned long __s[sizeof(long)==8?7:9]; } __u; } pthread_attr_t;
```

looking at the sourcecode of fallback.rs:817 this panic is due to some unsupported syntax.

what would be a quick & dirty workaround to fix this in proc-macro2?

thx!

**Comment by dtolnay** [comment]
Proc-macro2 is behaving correctly as far as I can tell. "pthread_attr_t_union_(unnamed_at_/usr/include/bits/alltypes_h_378_18)" is indeed not a valid identifier because none of '(' or '/' or ')' have unicode's XID_Continue property. You're looking for a bug that is somewhere else.

**Comment by dtolnay** [comment]
I published a change in proc-macro2 1.0.70 to give the caller's location in the panic message, instead of a location pointing to fallback.rs.

**Comment by stef** [comment]
very helpful! thank you!
